### PR TITLE
gcs: 4.8.0 -> 5.20.4, adopt, refactor

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

The `gcs` package has changed its build system over to `go` from `java/ant`

The original commit which added this package didn't set `meta.maintainers` so added myself to it.

I moved the package over to `pkgs/by-name`

I had to use a hack to get `mupdf` to be linked properly (weird upstream configs)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TomaSajt** [comment]
@ofborg build gcs

File: pkgs/by-name/gc/gcs/package.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    changelog = "https://github.com/richardwilkes/gcs/releases/tag/${src.rev}";
+    description = "A stand-alone, interactive, character sheet editor for the GURPS 4th Edition roleplaying game system";
+    homepage = "https://gurpscharactersheet.com/";
+    license = lib.licenses.mpl20;
+    mainProgram = "gcs";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```
Comment (kirillrdy) [inline pkgs/by-name/gc/gcs/package.nix:71]: unless there are cgo specific limitations, this can be removed and use default set by buildGoModule

Comment (TomaSajt) [reply pkgs/by-name/gc/gcs/package.nix:71]: https://github.com/richardwilkes/gcs/blob/master/build.sh#L102

Upstream has this limitatation, though I am not sure what are the exact reasons for it

Maybe it was that weird `mupdf` thingy with the `pdf` repo from the same account.
(Haven't checked the other deps, though)

Sadly, I don't have any other platforms than x86_64-linux

**Comment by TomaSajt** [comment]
Looks like a missed that there was another attempt at this same refactor
https://github.com/NixOS/nixpkgs/pull/252772/files

**Comment by wegank** [comment]
```
> chmod: cannot access 'vendor/github.com/richardwilkes/pdf': No such file or directory
```

**Comment by wegank** [comment]
@ofborg build gcs gcs.passthru.tests

**Comment by TomaSajt** [comment]
Thanks for the fixes

**Pull request merged**